torch/csrc/dynamo/guards.cpp

```diff
@@ -464,10 +465,125 @@ static PyObject* assert_size_stride(PyObject* dummy, PyObject* args) {
   Py_RETURN_TRUE;
 }
 
+typedef struct {
+  PyDictObject *dict;  // borrowed reference
+  uint64_t dict_version_tag;
+} AttrTag;
+
+static const char* module_guard_attrs[] = {
+  "_parameters",
+  "_buffers",
+  "_modules",
+  "_forward_hooks",
+  "_forward_pre_hooks",
+};
+
+typedef struct {
+  PyObject_HEAD
+  PyObject *wr;
+  unsigned int version_tag;
+  uint64_t dict_version_tag;
```

Comment:
Platform dependent and fixed sizes in the same struct?

Reply:
Those are the types of CPython's `tp_version_tag` and `ma_version_tag`. I don't make the rules!

https://github.com/python/cpython/blob/v3.10.11/Include/cpython/object.h#L268-L269

https://github.com/python/cpython/blob/v3.10.11/Include/cpython/dictobject.h#L16-L18